aptos-move/block-executor/src/executor.rs

```diff
@@ -136,38 +136,60 @@ where
         let mut updates_outside = false;
-        let mut apply_updates = |output: &E::Output| {
-            // First, apply writes.
-            for (k, v) in output
-                .resource_write_set()
-                .into_iter()
-                .chain(output.aggregator_v1_write_set().into_iter())
-            {
-                if prev_modified_keys.remove(&k).is_none() {
-                    updates_outside = true;
+        let mut apply_updates =
+            |output: &E::Output| {
+                // First, apply writes.
+                for (k, v) in output
+                    .resource_write_set()
+                    .into_iter()
+                    .chain(output.aggregator_v1_write_set().into_iter())
+                {
+                    if prev_modified_keys.remove(&k).is_none() {
+                        updates_outside = true;
+                    }
+                    versioned_cache
+                        .data()
+                        .write(k, idx_to_execute, incarnation, v);
                 }
-                versioned_cache
-                    .data()
-                    .write(k, idx_to_execute, incarnation, v);
-            }
 
-            for (k, v) in output.module_write_set().into_iter() {
-                if prev_modified_keys.remove(&k).is_none() {
-                    updates_outside = true;
+                for (k, v) in output.module_write_set().into_iter() {
+                    if prev_modified_keys.remove(&k).is_none() {
+                        updates_outside = true;
+                    }
+                    versioned_cache.modules().write(k, idx_to_execute, v);
                 }
-                versioned_cache.modules().write(k, idx_to_execute, v);
-            }
 
-            // Then, apply deltas.
-            for (k, d) in output.aggregator_v1_delta_set().into_iter() {
-                if prev_modified_keys.remove(&k).is_none() {
-                    updates_outside = true;
+                // Then, apply deltas.
+                for (k, d) in output.aggregator_v1_delta_set().into_iter() {
+                    if prev_modified_keys.remove(&k).is_none() {
+                        updates_outside = true;
+                    }
+                    versioned_cache.data().add_delta(k, idx_to_execute, d);
+                }
+
+                for (id, change) in output.aggregator_v2_change_set().into_iter() {
+                    if !prev_modified_aggregators.remove(&id) {
+                        updates_outside = true;
+                    }
+
+                    // TODO: figure out if change should update updates_outside
+                    // versioned_cache.aggregators().
+                    match change {
+                        AggregatorChange::Create(value) => versioned_cache
+                            .aggregators()
+                            .create_aggregator(id, idx_to_execute, value),
+                        AggregatorChange::Apply(apply) => versioned_cache
```

Comment:
I guess record_apply can return a boolean to determine if updates_outside must be set, although at that point updates_outside isn't a good name anymore.